pulumi/infra/postgres.py

```diff
@@ -0,0 +1,115 @@
+from typing import List, Optional
+
+import pulumi_aws as aws
+import pulumi_random as random
+
+import pulumi
+
+
+class Postgres(pulumi.ComponentResource):
+    """
+    A Postgres instance running in RDS.
+    """
+
+    def __init__(
+        self,
+        name: str,
+        vpc_id: pulumi.Input[str],
+        subnet_ids: pulumi.Input[List[str]],
+        nomad_agent_security_group_id: pulumi.Input[str],
+        opts: Optional[pulumi.ResourceOptions] = None,
+    ) -> None:
+        super().__init__("grapl:Postgres", name, None, opts)
+
+        # Burstable, 2GB memory, Gravitron 2.
+        # With 5GB of storage and all-day usage, it comes out to about $24/mo
+        instance_class = "db.t4g.small"
+
+        child_opts = pulumi.ResourceOptions(parent=self)
+
+        self.username = "postgres"
+        self.password = random.RandomPassword(
+            "password",
+            length=32,
+            # Disable special characters, ":" can lead to unexpected
+            # sqlx errors: https://github.com/launchbadge/sqlx/issues/1624
+            special=False,
+            opts=child_opts,
+        ).result
+
+        rds_subnet_group = aws.rds.SubnetGroup(
+            "subnet-group",
+            subnet_ids=subnet_ids,
+            tags={"Name": f"{name}-subnet-group"},
+            opts=child_opts,
+        )
+
+        self.security_group = aws.ec2.SecurityGroup(
+            f"postgres-security-group",
```

Comment:
`f"{name}-security-group"`

(Or maybe `f"{name}-postgres-security-group"`, if we're adding that as a `Name` tag below? If so, other resource names should probably include `postgres` as well)